source/common/router/config_impl.cc

```diff
@@ -290,6 +290,7 @@ RouteEntryImplBase::RouteEntryImplBase(const VirtualHostImpl& vhost,
                                                       route.request_headers_to_remove())),
       response_headers_parser_(HeaderParser::configure(route.response_headers_to_add(),
                                                        route.response_headers_to_remove())),
+      metadata_(route.metadata()), typed_metadata_(route.metadata()),
```

Comment:
Looking at this, I feel more convinced that we should always return the typed metadata variant instead of storing two forms.. ITs 2x the storage.

Reply:
> Looking at this, I feel more convinced that we should always return the typed metadata variant instead of storing two forms.. ITs 2x the storage.

Not necessarily, typed_metadata_ only contains data entries that has a translation registered.
For "plain" data entries like of primitive values, I'd think there is less drive to wrap them in another class.